xqueue_watcher/client.py

```diff
@@ -63,7 +63,14 @@ def _request(self, method, uri, timeout=0.5, **kwargs):
         r = None
         while not r:
             try:
-                r = self.session.request(method, url, auth=self.http_basic_auth, timeout=timeout, **kwargs)
+                r = self.session.request(
+                    method,
+                    url,
+                    auth=self.http_basic_auth,
+                    timeout=timeout,
+                    allow_redirects=False,
+                    **kwargs
+                )
```

Comment:
Seconded. I think a bare-er `void loop() {}` type function is nicer. But it's a nit.